analytics/src/FirebaseAnalytics.cs

```diff
@@ -248,6 +251,59 @@ public static void SetConsent(System.Collections.Generic.IDictionary< ConsentTyp
     FirebaseAnalyticsInternal.SetConsentWithInts(consentSettingsMap);
   }
 
+  /// @brief Sets default event parameters.
+  ///
+  /// These parameters are logged with all events, in addition to the parameters passed to the
+  /// LogEvent() call. They are useful for logging common parameters with all events.
+  /// Default event parameters are overridden by event-specific parameters if the names are the
+  /// same. Default parameters are removed if the dictionary is null or empty.
+  ///
+  /// @param[in] parameters The dictionary of parameters to set.
+  ///   If null, clears all default parameters.
+  public static void SetDefaultEventParameters(
+      System.Collections.Generic.IDictionary<string, object> parameters) {
+    if (parameters == null || parameters.Count == 0) {
+      // This handles both an explicitly null dictionary and an empty one.
+      FirebaseAnalyticsInternal.ClearDefaultEventParameters();
+      logger.LogMessage(LogLevel.Debug, "Cleared default event parameters.");
+    } else {
+      StringList parameterNames = new StringList();
+      VariantList parameterValues = new VariantList();
+      int originalCount = parameters.Count; // Store original count for later check
+
+      foreach (var kvp in parameters) {
+        try {
+          parameterNames.Add(kvp.Key);
+          parameterValues.Add(Firebase.Variant.FromObject(kvp.Value));
+        } catch (System.Exception e) {
+          logger.LogMessage(LogLevel.Warning, string.Format(
+              "Failed to convert default parameter '{0}'. Skipping. Error: {1}",
+              kvp.Key, e.ToString()));
+          // If adding to parameterNames succeeded but Variant.FromObject failed,
+          // we need to remove the key that was added optimistically.
+          if (parameterNames.Count > parameterValues.Count) {
+              parameterNames.RemoveAt(parameterNames.Count - 1);
+          }
+        }
+      }
+
+      if (parameterNames.Count == 0 && originalCount > 0) {
```

Comment:
I don't think we need this originalCount logic any more if we change it so an empty dictionary is just passed through verbatim.